### PR TITLE
Initial MVC model

### PR DESCRIPTION
**Pull request by TimBartholomew**
## Fixes/Addresses:
#295

## Summary/Motivation:
WaterTAP needs an MVC model

## Changes proposed in this PR:
- adds multiple unit models that compose the mvc process
- adds a property package for just water vapor and liquid
- adds a call back and a test for a heat exchanger with chen's approximation
- adds a compressor model that has a different formulation than IDAES
- adds a full condenser model
- adds an evaporator model
- adds a test combining and solving all of the linked units

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/unit_models/mvc/components/tests/test_evaporator.py [hunk 1]
```diff
@@ -29,6 +30,7 @@
 solver = get_solver()
 
 # -----------------------------------------------------------------------------
+@pytest.mark.skipif(sys.platform.startswith("darwin"), reason="Evaporator needs complied IDAES extensions")
```
Comment (lbianchi-lbl) [inline watertap/unit_models/mvc/components/tests/test_evaporator.py]: Just FYI, the predefined `@pytest.mark.requires_idaes_solver` marker can be used for this functionality (skipping/xfailing tests if IDAES extensions are not available).

File: watertap/unit_models/mvc/tests/test_mvc.py [hunk 2]
```diff
@@ -99,7 +100,7 @@ def initialize(m):
     propagate_state(m.fs.s01)
     m.fs.compressor.initialize(outlvl=idaeslog.INFO_HIGH)
 
-
+@pytest.mark.skipif(sys.platform.startswith("darwin"), reason="MVC needs complied IDAES extensions")
```
Comment (lbianchi-lbl) [inline watertap/unit_models/mvc/tests/test_mvc.py]: See above for possible use case for `@pytest.mark.requires_idaes_solver`

**Review by adam-a-a — APPROVED** [review]
While there are a few things that we might consider refining in a future PR, I think this looks good for the initial implementation.

**Comment by OOAmusat** [comment]
Looks good.

**Review by OOAmusat — APPROVED** [review]
Looks good.

**Pull request merged**